eva_vcf_merge/detect.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2021 EMBL - European Bioinformatics Institute
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import itertools
+from enum import Enum
+
+import pysam
+
+from eva_vcf_merge.utils import are_all_elements_unique
+
+
+class SampleSetType(Enum):
+    SINGLE_SET = 'single set'
+    UNSORTED_SINGLE_SET = 'unsorted single set'
+    UNIQUE_SETS = 'unique sets'
+    OVERLAPPING_SETS = 'overlapping sets'
+
+
+class MergeType(Enum):
+    HORIZONTAL = 'horizontal'
+    VERTICAL = 'vertical'
+
```

Comment:
More of a nice to have: a brief description of each of the SampleSetType and also an illustration (or a URL to an existing one) of horizontal and vertical merging.

Reply:
I added some description in the code and some (very simple) illustration to the [readme](https://github.com/apriltuesday/eva-vcf-merge/blob/EVA-2366/README.md), can you let me know if it seems useful?

If we want to make this really useable by other people, we will probably need more examples and diagrams that aren't just colourful boxes, and some detail on the issues you describe in your [other comment](https://github.com/EBIvariation/eva-vcf-merge/pull/2/files#r685468202)... but I guess we have to start somewhere.

Reply:
Wow! I was just looking for a poor-man's stick diagram. This is way better!! Thanks!